### ISSUE TITLE
BUG - sev-3 - All - Better error handling when a message is older than 45 days and contains a reply draft

### ISSUE THREAD

**Issue by TKDickson** [department-of-veterans-affairs/va-mobile-app#7202]
## What happened?

Right now, if a user has a reply draft for a message that's older than 45 days and attempts to finish and send the draft, they get the generic messaging error snackbar which gives them the option to try again. They will also get the generic error snackbar when attempting to save the draft, for the same reason, which also has a try again button.

However, messages older than 45 days are not allowed to be replied to, so trying again will never work in either case -- we're getting a more specific error message from the backend about it, so we can do better error handling to let the veteran know to start a new message/pick a different path.

## Specs:

- Device:
- OS:
- User Account: +226's 7/6/2023 draft "Medication: Medication Inquiry" will show this behavior
- Accessibility State:

## Steps to Reproduce

See Charles JSON in details section

## Desired behavior

Website. Not great, but technically better than us:
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeE9oQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--9beb273de9fa83c836b36406fdc7e4f6595f11f8/image.png)

UX (and maybe content?) will need to weigh in on our desired behavior.

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: Low 
- Frequency: Low 

- 3 - Low

## Screen shot(s) and additional information

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeU9oQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--9d238baf4f5c0ba9c4f891de1e2b0c47422e9656/image.png)

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  
POST /mobile/v0/messaging/health/messages/2895789/reply HTTP/1.1
Host: staging-api.va.gov
x-key-inflection: camel
Accept: */*
Authorization: Bearer eyJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJ2YS5nb3Ygc2lnbiBpbiIsImF1ZCI6InZhbW9iaWxlIiwiY2xpZW50X2lkIjoidmFtb2JpbGUiLCJqdGkiOiJiY2U1OTFhZS1hMTU0LTRjNjYtYWE1NS0yN2ZjODRjZjgwM2EiLCJzdWIiOiIxMzA2ZTMxMjczNjA0ZGQ0YTEyYWE2NzYwOWE2M2JmZSIsImV4cCI6MTY5OTAzODAwMCwiaWF0IjoxNjk5MDM2MjAwLCJzZXNzaW9uX2hhbmRsZSI6IjllMWQ2MDU0LWJkNTctNGQxZS1hMGVhLTI3MWUyNzc0MGQyYSIsInJlZnJlc2hfdG9rZW5faGFzaCI6ImMyNTc5OTIwZmY2M2YxYzY0ZDgxZjMzMzgzNDU0ZjFmYzhhZjllZGFkN2FlM2E2MWZjMTZkOTMwZjI4NGQwMDkiLCJwYXJlbnRfcmVmcmVzaF90b2tlbl9oYXNoIjoiZDg5MDE5ZjRmZTM2MTEzMzc5OTIyZjIzYmYyNWQ5OTk2NmUyNDQxZDUxZTI5ZmJkN2ExOTI4NmE3MmVkMTUyNyIsImFudGlfY3NyZl90b2tlbiI6ImEyZTg1MWE3NjFhYzFmOTQ1ZTY5ZTNkNmYzNjQ1NTEwIiwibGFzdF9yZWdlbmVyYXRpb25fdGltZSI6MTY5OTAzNjE5OSwidmVyc2lvbiI6IlYwIiwidXNlcl9hdHRyaWJ1dGVzIjp7fX0.SOaFyI0J9kO0-cmRK83b-A43jDA06GqbWdc3MhYdPPKwzVXuLwzrKpTfzQLRQATRuqlhM7Jw-KIKQ7-2RZmVZLBNWpFVQw1WhxMN_pH2rDbtCxpF_da2ICARMQJ32M7X2phvm8NjCbpm7Xg28g6WvObVDbmM1W-2CZFw8O6cKnJDVqfbLS7_dtCi4lF8KaMNhR5RXVeRzhc3bQudpcJMZmW4gmQkmGVXJqTPtr-fIOpgmonhsK-CNLlcP_echXLV9dGwm6l9B_m8Ohr4qahxGMLI864X8vnuO_28F5-RzFhJ3F2u4J9U1nAusHSZANMtxVzWxtVuOI6owkTi-CipAQ
authentication-method: SIS
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
Content-Type: application/json
Content-Length: 61
User-Agent: VAMobile/1045 CFNetwork/1404.0.5 Darwin/22.3.0
Connection: keep-alive
source-app-name: va-health-benefits-app

{"body":"Test\n","draft_id":2887719,"category":"MEDICATIONS"}

{
	"errors": [{
		"title": "Operation failed",
		"detail": "Unable to reply because the source message is expired",
		"code": "SM130",
		"status": "422"
	}]
}

PUT /mobile/v0/messaging/health/message_drafts/2895789/replydraft/2887719 HTTP/1.1
Host: staging-api.va.gov
x-key-inflection: camel
Accept: */*
Authorization: Bearer eyJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJ2YS5nb3Ygc2lnbiBpbiIsImF1ZCI6InZhbW9iaWxlIiwiY2xpZW50X2lkIjoidmFtb2JpbGUiLCJqdGkiOiJiY2U1OTFhZS1hMTU0LTRjNjYtYWE1NS0yN2ZjODRjZjgwM2EiLCJzdWIiOiIxMzA2ZTMxMjczNjA0ZGQ0YTEyYWE2NzYwOWE2M2JmZSIsImV4cCI6MTY5OTAzODAwMCwiaWF0IjoxNjk5MDM2MjAwLCJzZXNzaW9uX2hhbmRsZSI6IjllMWQ2MDU0LWJkNTctNGQxZS1hMGVhLTI3MWUyNzc0MGQyYSIsInJlZnJlc2hfdG9rZW5faGFzaCI6ImMyNTc5OTIwZmY2M2YxYzY0ZDgxZjMzMzgzNDU0ZjFmYzhhZjllZGFkN2FlM2E2MWZjMTZkOTMwZjI4NGQwMDkiLCJwYXJlbnRfcmVmcmVzaF90b2tlbl9oYXNoIjoiZDg5MDE5ZjRmZTM2MTEzMzc5OTIyZjIzYmYyNWQ5OTk2NmUyNDQxZDUxZTI5ZmJkN2ExOTI4NmE3MmVkMTUyNyIsImFudGlfY3NyZl90b2tlbiI6ImEyZTg1MWE3NjFhYzFmOTQ1ZTY5ZTNkNmYzNjQ1NTEwIiwibGFzdF9yZWdlbmVyYXRpb25fdGltZSI6MTY5OTAzNjE5OSwidmVyc2lvbiI6IlYwIiwidXNlcl9hdHRyaWJ1dGVzIjp7fX0.SOaFyI0J9kO0-cmRK83b-A43jDA06GqbWdc3MhYdPPKwzVXuLwzrKpTfzQLRQATRuqlhM7Jw-KIKQ7-2RZmVZLBNWpFVQw1WhxMN_pH2rDbtCxpF_da2ICARMQJ32M7X2phvm8NjCbpm7Xg28g6WvObVDbmM1W-2CZFw8O6cKnJDVqfbLS7_dtCi4lF8KaMNhR5RXVeRzhc3bQudpcJMZmW4gmQkmGVXJqTPtr-fIOpgmonhsK-CNLlcP_echXLV9dGwm6l9B_m8Ohr4qahxGMLI864X8vnuO_28F5-RzFhJ3F2u4J9U1nAusHSZANMtxVzWxtVuOI6owkTi-CipAQ
authentication-method: SIS
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
Content-Type: application/json
Content-Length: 61
User-Agent: VAMobile/1045 CFNetwork/1404.0.5 Darwin/22.3.0
Connection: keep-alive
source-app-name: va-health-benefits-app

{"body":"Test\n","draft_id":2887719,"category":"MEDICATIONS"}

{
	"errors": [{
		"title": "Operation failed",
		"detail": "Unable to reply because the source message is expired",
		"code": "SM130",
		"status": "422"
	}]
}

</details>

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the [bug epic](https://app.zenhub.com/workspaces/va-mobile-product-view-610035bc5395bb000e62e529/issues/gh/department-of-veterans-affairs/va-mobile-app/5367)

**Comment by bischoffa** [comment]
Added ticket dependency for UX to work if / when prioritized by appropriate team.

**Comment by Sparowhawk** [comment]
decision was to apply the you can't reply to this message 45 day alert, and change the reply/send message button to the start new message button

**Comment by alexandec** [comment]
@rbontrager I just created [a PR for this](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/8348) which also changes the demo mode data a bit. I had to change two of the messages so they use dynamic dates. That way we have a message and a draft reply to the message, that both fall into the 45-day window. I'm sure some E2E test changes will be needed but I'm not sure which of us should make the E2E changes.
